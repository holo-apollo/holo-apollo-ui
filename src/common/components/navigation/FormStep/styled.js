// @flow
import styled, { css } from 'styled-components';
import palette from 'common/palette';

type StyledProps = {
  isCurrent: boolean,
  isDisabled: boolean,
};

const getContStyles = ({ isCurrent, isDisabled }: StyledProps) => css`
  background-color: ${isCurrent ? palette.evening : palette.lightestGrey};
  color: ${isCurrent
    ? palette.white
    : isDisabled
    ? palette.lightGrey
    : palette.darkGrey};
  cursor: ${isDisabled ? 'not-allowed' : 'pointer'};
`;

export const Cont = styled.div`
  height: 45px;
  display: flex;
  align-items: center;
  justify-content: center;
  ${getContStyles};
`;

export const ContentCont = styled.div`
  text-align: center;
`;

export const HeaderCont = styled.div`
  font-size: 14px;
  font-weight: bold;
  text-transform: uppercase;
`;

export const HelpTextCont = styled.div`
  font-size: 9px;
`;
