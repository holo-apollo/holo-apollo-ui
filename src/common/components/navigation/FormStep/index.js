// @flow
import React from 'react';

import { Cont, ContentCont, HeaderCont, HelpTextCont } from './styled';

type Props = {
  isCurrent: boolean,
  isDisabled: boolean,
  header: string,
  helpText: string,
  onClick: () => void,
};

const FormStep = ({
  isCurrent,
  isDisabled,
  header,
  helpText,
  onClick,
}: Props) => (
  <Cont isCurrent={isCurrent} isDisabled={isDisabled} onClick={onClick}>
    <ContentCont>
      <HeaderCont>{header}</HeaderCont>
      <HelpTextCont>{helpText}</HelpTextCont>
    </ContentCont>
  </Cont>
);

export default FormStep;
