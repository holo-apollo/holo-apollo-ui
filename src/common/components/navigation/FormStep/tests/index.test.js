import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import FormStep from '../index';

describe('FormStep', function() {
  it('should render normally', function() {
    const wrapper = mount(
      <FormStep
        isCurrent={true}
        isDisabled={false}
        header="Header"
        helpText="Help text"
        onClick={jest.fn()}
      />
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should match snapshot', function() {
    const tree = renderer
      .create(
        <FormStep
          isCurrent={true}
          isDisabled={false}
          header="Header"
          helpText="Help text"
          onClick={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
