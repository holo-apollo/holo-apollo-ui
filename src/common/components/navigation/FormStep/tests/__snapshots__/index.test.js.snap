// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStep should match snapshot 1`] = `
<div
  className="styled__Cont-sc-150835y-0 btLMjA"
  onClick={[MockFunction]}
>
  <div
    className="styled__ContentCont-sc-150835y-1 gxYQRH"
  >
    <div
      className="styled__HeaderCont-sc-150835y-2 iabCzy"
    >
      Header
    </div>
    <div
      className="styled__HelpTextCont-sc-150835y-3 jYzyzO"
    >
      Help text
    </div>
  </div>
</div>
`;
